full-service/src/service/transaction_builder.rs

```diff
@@ -268,7 +268,10 @@ impl<FPR: FogPubkeyResolver + 'static> WalletTransactionBuilder<FPR> {
         Ok(FullServiceFogResolver(fully_validated_fog_pubkeys))
     }
 
-    pub fn build_unsigned(&self) -> Result<UnsignedTx, WalletTransactionBuilderError> {
+    pub fn build_unsigned(
+        &self,
+        redemption_memo: Option<[u8; BurnRedemptionMemo::MEMO_DATA_LEN]>,
```

Comment:
To future-proof this for other memos it might be worth making this an enum that currently only has one variant (the redemption memo). Otherwise for the next memo we want to use in an unsigned transaction another argument will need to be added, as well as another field to `UnsignedTx` - which then also makes it possible to provide two memos which is an error case.

Reply:
Ah, good point. Will implement something like that. Thanks!